tests/test_util.py

```diff
@@ -142,8 +158,444 @@ def test_IsAtTrajectoryEnd_ReturnsTrue(self):
 
     def test_IsAtTrajectoryEnd_ReturnsFalse(self):
         current_config = self.robot.GetDOFValues(self.active_dof_indices)
-        goal_config = [0.02, 0.01, 0.02, 0.01, 0.01, 0.01, 0.0];
+        goal_config = [0.02, 0.01, 0.02, 0.01, 0.01, 0.01, 0.0]
         q0 = current_config
         q1 = goal_config
         traj = self.CreateTrajectory(q0, q1)
         self.assertFalse(prpy.util.IsAtTrajectoryEnd(self.robot, traj))
+
+
+    # CheckJointLimits()
+    #
+    # Note: the WAM arm joint limits are:
+    #       q_limit_min =
+    #           [-2.617, -1.972, -2.740, -0.872, -4.799, -1.570, -3.001]
+    #       q_limit_max =
+    #           [ 2.617,  1.972,  2.740,  3.141,  1.308,  1.570,  3.001]
+
+    def test_CheckJointLimits_MinLimitOneJoint(self):
+        # One joint is beyond the min limit
+        q0 = [0.0, 0.0, 0.0, -1.0, 0.0, 0.0, 0.0]
+        try:
+            prpy.util.CheckJointLimits(self.robot, q0)
+        except JointLimitError:
+            pass # test passed
+        except Exception, e:
+            error = 'Unexpected exception thrown: ' + str(e.message)
+            self.fail(error)
+        else:
+            self.fail('Expected exception not thrown')
```

Comment:
The `unittest` module has built-in functionality for doing this:
```python
with self.assertRaises(JointLimitError):
    prpy.util.CheckJointLimits(self.robot, q0)
```

This comment applies to a bunch of other tests, too.